org.eclipse.dartboard/src/org/eclipse/dartboard/preference/DartSDKLocationFieldEditor.java

```diff
@@ -17,15 +27,59 @@ public DartSDKLocationFieldEditor(String preferencesKey, String label, Composite
 	@Override
 	protected boolean doCheckState() {
 		String location = getTextControl().getText();
-		File file = new File(location);
-		boolean isValid = file.exists();
+		boolean isValid = isValidDartSDK(location);
 		if (!isValid) {
 			setErrorMessage(Messages.Preference_SDKNotFound_Message);
 			showErrorMessage();
 		}
 		return isValid;
 	}
 
+	@SuppressWarnings("nls")
+	private boolean isValidDartSDK(String location) {
+		Path path = Paths.get(location);
+		// If the entered file doesn't exist, there is no need to run it
+		// Similarly if the file is not a directory
+		if (!Files.exists(path)) {
+			return false;
+		}
+		try {
+			path = path.toRealPath();
+		} catch (IOException e1) {
+			LOG.error("Couldn't follow symlink", e1);
+			return false;
+		}
+
+		String executablePath = null;
+		if (path.endsWith("bin/dart")) {
+			executablePath = path.toAbsolutePath().toString();
+		} else if (path.endsWith("bin")) {
+			executablePath = path.toAbsolutePath().toString() + "/dart";
+		} else {
+			executablePath = path.toAbsolutePath().toString() + "/bin/dart";
+		}
+
+		String[] commands;
+		if (Platform.OS_WIN32.equals(Platform.getOS())) {
```

Comment:
We can write  like this also		
String[] commands =  PlatformUtil.isWindows()? new String[] {"cmd", "/c", executablePath + " --version": new String[] {"/bin/bash", "-c", executablePath + " --version"};

add a method for OS check in PlatformUtil class.

Reply:
This ternary operator makes it very hard to read. I like the If-else better.

We don't have a PlatformUtil class yet. Only PlatformUIUtil which doesn't seem to fit for this case. I don't like adding all the util classes with only one or two util method. For now I think this is fine. In the future we could also use a utility method to just pass a command to a function and it then executes the appropriate process.. But that's out of scope of this change IMO.

Does this work on Windows btw?

Reply:
Seems to be validation broken on windows. I can see validation error for valid installation.

Reply:
Yeah, I thought so. Could you check how we would have to execute a command with cmd on windows? The command should look like this: `cmd /c C:\path\to\dart --version`

Reply:
> 'C:\Program' is not recognized as an internal or external command,

Reply:
Change to 			commands = new String[] { "cmd", executablePath + " --version" };

Reply:
Something wrong in the windows side. It's returned windows version  

> Microsoft Windows [Version 10.0.18362.239]

Reply:
> Something wrong in the windows side. It's returned windows version

What do you mean? Does `Platform.OS_WIN32.equals(Platform.getOS())` evaluate to false?

Reply:
No. command  has the wrong syntax
Recommended code for windows:

```
		boolean isWindows = Platform.OS_WIN32.equals(Platform.getOS());
		String dartFile = isWindows ? "dart.exe" : "dart";
		String executablePath = null;
		if (path.endsWith("bin" + File.separator + dartFile)) {
			executablePath = path.toAbsolutePath().toString();
		} else if (path.endsWith("bin")) {
			executablePath = path.toAbsolutePath().toString() + File.separator + dartFile;
		} else {
			executablePath = path.toAbsolutePath().toString() + File.separator + "bin" + File.separator + dartFile;
		}

		String[] commands;
		if (isWindows) {
			commands = new String[] { "cmd", "/c", executablePath, "--version" };
		} else {
			commands = new String[] { "/bin/bash", "-c", executablePath + "--version" };
		}


```

Reply:
I removed the part with the if-else if-else because it allowed a few special locations through that weren't actual Dart SDK locations. I also added the correct syntax for windows. Please check again, if it works.